src/nspawn/nspawn.c

```diff
@@ -378,6 +379,33 @@ static void parse_share_ns_env(const char *name, unsigned long ns_flag) {
         arg_clone_ns_flags = (arg_clone_ns_flags & ~ns_flag) | (r > 0 ? 0 : ns_flag);
 }
 
+static void parse_mount_settings_env(void) {
+        int r;
+        const char *e;
+
+        e = getenv("SYSTEMD_NSPAWN_API_VFS_WRITABLE");
+        if (!e)
+                return;
+
+        if (streq(e, "network")) {
+                arg_mount_settings |= MOUNT_APPLY_APIVFS_RO;
+                arg_mount_settings |= MOUNT_APPLY_APIVFS_NETNS;
+                return;
+        }
+
+        r = parse_boolean(e);
+
+        if (r < 0) {
+                log_warning_errno(r, "Failed to parse SYSTEMD_NSPAWN_API_VFS_WRITABLE from environment, ignoring.");
+                return;
+        } else if (r > 0) {
+                arg_mount_settings &= ~MOUNT_APPLY_APIVFS_RO;
+                arg_mount_settings &= ~MOUNT_APPLY_APIVFS_NETNS;
+        } else {
+                arg_mount_settings |= MOUNT_APPLY_APIVFS_RO;
```

Comment:
I am not sure I follow the discussion now. I think the code should be backwards compat when the env var is not set, but do exactly what the env var says when it is set. That means: if the env var is not set then we mount /proc/sys and /sys read-only by default, except for /proc/sys/net — but the latter only if private networking is on. However, if the env var is set to yes, no or network, then the private networking setting should not have any effect anymore, and what the user asked for should take effect. That means: if set to "yes", then all of /proc/sys and /sys should be writable. If set to "no", then nothing of it should be writable, and if set to "network", then /proc/sys + /sys should be read-only, but /proc/sys/net should be writable. 

Or in other words: I think the env var should turn off any kind of automatism, and permit the caller to *force* the invocation into a certain mode. 

And thus I think the MOUNT_APPLY_APIVFS_NETNS flag should be turned off in this branch...

Or am I missing something?